SingularityService/src/main/java/com/hubspot/singularity/resources/TaskResource.java

```diff
@@ -180,6 +183,24 @@ public SingularityTaskRequest getPendingTask(@Auth SingularityUser user, @PathPa
     return taskRequestList.get(0);
   }
 
+  @DELETE
+  @Path("/scheduled/task/{scheduledTaskId}")
+  @ApiOperation("Delete a scheduled task.")
+  public Void deleteScheduledTask(@Auth SingularityUser user, @PathParam("scheduledTaskId") String taskId, @Context HttpServletRequest requestContext) {
+    return maybeProxyToLeader(requestContext, Void.class, null, () -> deleteScheduledTask(taskId, user));
+  }
+
+  private Void deleteScheduledTask(String taskId, SingularityUser user) {
+    SingularityTaskRequest taskRequest = getPendingTask(user, taskId);
```

Comment:
`getPendingTask` is the other resource method on this class that does it's own separate checks and validations before returning a value. Rather than relying on that endpoint's checks, we should prefer interacting directly with task manager here. The situations in which each endpoint are used can vary greatly, so we would generally prefer interacting with the data vs interacting with another 'endpoint' (even though that endpoint here is just another method not an api call). This way our new endpoint can manage its own behavior for situations like when the pending task is not found, and not have to worry about catching and processing exceptions from the other endpoint to determine what happened.

`taskManager` has a `getPendingTask` as well, that will return an Optional, making our check for a pending task that wasn't found easier. Also, related to the comment above about return type, we can then more easily return an `Optional<SingularityPendingTaskId>` in the case where the pending task didn't exist. (As a note, jax-rs automatically maps an absent/empty Optional to a 404, whereas a `Void` will be mapped to a 204)

Also, we don't need the full `SingularityTaskRequest` data, so calling `TaskResource.getPendingTask` is doing extra work. The authorization helper has a `checkForAuthorizationByRequestId` method, and the pending task id has the request id present in it. So, we don't actually need the extra task requests call that `TaskResource.getPendingTask` is doing.

Reply:
I think we still need the `SingularityTaskRequest` because we're checking ON_DEMAND type in addition to auth. Is there another way to check type without that?

Reply:
Ah, forgot about the type check. You're correct we need more info. We need the SingularityRequest, but not necessarily the SingularityDeploy (which also comes with the task request and can be a lot more data). You can use the requestManager to grab that